Implement Format for str

This code
```rust
        let s: &str = "hello";
        info!("my string is {:?}", s);
```

produces this error

```
error[E0277]: the trait bound `str: Format` is not satisfied
   --> foobar.rs:292:9
    |
292 |         info!("my string is {:?}", s);
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Format` is not implemented for `str`
    |
    = note: required because of the requirements on the impl of `Format` for `&str`
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Not a huge issue because there are better ways to print strings, using `intern!()` or with `{:str}`. However being able to use `{:?}` can come in handy when writing macros or generic functions that need to log arbitrary types that could potentially be `str`.

--- a/firmware/qemu/src/bin/log.rs
+++ b/firmware/qemu/src/bin/log.rs
@@ -81,6 +81,7 @@ fn main() -> ! {
 
     let interned = defmt::intern!("interned string");
     defmt::info!("&str = {:str}", "string slice");
+    defmt::info!("&str = {:?}", "string slice");
     defmt::info!("&Str = {:istr}", interned);
     defmt::info!("&Str = {:?}", interned);
 

--- a/src/impls.rs
+++ b/src/impls.rs
@@ -110,6 +110,16 @@ impl Format for f32 {
     }
 }
 
+impl Format for str {
+    fn format(&self, fmt: &mut Formatter) {
+        if fmt.needs_tag() {
+            let t = internp!("{:str}");
+            fmt.u8(&t);
+        }
+        fmt.str(self);
+    }
+}
+
 impl Format for Str {
     fn format(&self, fmt: &mut Formatter) {
         if fmt.needs_tag() {

